src/components/TopBar/TopBar.jsx

```diff
@@ -1,2 +1,115 @@
-import React from "react";
-export const TopBar = () => <>TopBar</>;
+import { React, useState } from "react";
+import {
+  StyledTopBarWrapper,
+  StyledTitle,
+  StyledToolsWrapper,
+  StyledSvg,
+  StyledLineSvg,
+  StyledSvgWrapper,
+  StyledName,
+  StyledImgBorder,
+  StyledImg,
+  StyledInput,
+} from "./style";
+import { useLocation } from "react-router-dom";
+import profilePhoto from "../../images/profile-photo.png";
+
+export const TopBar = () => {
+  const location = useLocation();
+  const [isInputDisplayed, setIsInputDisplayed] = useState(false);
+
+  const handleSearchClick = () => {
+    setIsInputDisplayed(!isInputDisplayed);
+  };
+
+  const getTitle = () => {
```

Comment:
I probably would suggest to move this function to helpers folder near, it is big enough and could be used as pure function. Also pure functions are a priority if there is not too many used outside variables, so they can be provided as function params